### PR TITLE
Implement table serialize and deserialize

### PR DESCRIPTION
**Pull request by duanyytop**

### CODE REVIEW

**Comment by Keith-CY** [comment]
Test failed

**Comment by duanyytop** [comment]
> Test failed

Fixed.

File: src/utils/index.ts [hunk 1]
```diff
@@ -18,6 +18,16 @@ export const uint32Le = (uint32: string) => {
   return `0x${[...new Uint8Array(buf)].map(v => v.toString(16).padStart(2, '0')).join('')}`
 }
 
+export const littleHexToInt = (littleHex: string) => {
+  assertIsHexStr(littleHex)
```
Comment (Keith-CY) [inline src/utils/index.ts]: It's possible to use https://github.com/Keith-CY/molecule-javascript/pull/8/files#diff-2cfb1554a9cbe2bf7981031a26f82e4eR41 here.

File: src/table/index.ts [hunk 2]
```diff
@@ -1 +1,40 @@
...
+import { deserializeArray } from '../array'
+
+const Uint32Length = 4
+
+export const serializeTable = (origin: Array<TypeElement>): string => {
+  const values = origin.map(item => item[1])
+  values.forEach(item => {
+    assertIsHexStr(item)
+  })
+  values.map(item => (item.length % 2 ? `0${item.slice(2)}` : item.slice(2)))
```
Comment (Keith-CY) [inline src/table/index.ts]: Should the result be reassigned to `values`?

File: src/table/index.ts [hunk 3]
```diff
@@ -1 +1,40 @@
...
+  return result
+}
+
+export const deserializeTable = (serialized: string, sizes: TypeSize[]) => {
+  assertIsHexStr(serialized)
+  const valuesLength = sizes.map(size => size[1]).reduce((total, value) => total + value)
+  const offsetsLength = (1 + sizes.length) * Uint32Length
+  const lengthHex = serialized.slice(0, 10)
+  if (littleHexToInt(lengthHex) !== valuesLength + offsetsLength) {
+    throw new TypeError(`Expect sum of size to be correct`)
```
Comment (Keith-CY) [inline src/table/index.ts]: It seems not to be a TypeError

File: src/utils/index.ts (hunk 1 above)
Comment (duanyytop) [reply src/utils/index.ts]: Updated. https://github.com/Keith-CY/molecule-javascript/pull/8/commits/c4361b5cfa7ac6b3849a29b3dd7c81410ee33a35

File: src/table/index.ts (hunk 2 above)
Comment (duanyytop) [reply src/table/index.ts]: Updated. https://github.com/Keith-CY/molecule-javascript/pull/8/commits/c4361b5cfa7ac6b3849a29b3dd7c81410ee33a35

File: src/table/index.ts (hunk 3 above)
Comment (duanyytop) [reply src/table/index.ts]: Updated. https://github.com/Keith-CY/molecule-javascript/pull/8/commits/c4361b5cfa7ac6b3849a29b3dd7c81410ee33a35

**Comment by duanyytop** [comment]
Add two table test cases in https://github.com/Keith-CY/molecule-javascript/pull/8/commits/f1c035d96ab58e5fd20c092046605329a1a519b6

**Pull request merged**